### PR TITLE
add region flag

### PR DESCRIPTION
**Pull request by nph4rd**
depends on: https://github.com/agentsea/agentdesk/pull/26

### CODE REVIEW

**Review by zhuraromdev — APPROVED** [review]

**Comment by pbarker** [comment]
Okay update here, I've made the image available and public in the following regions:

```
"us-west-1" "us-west-2" "eu-west-1" "eu-central-1" "ap-southeast-1"
```

We should add a check that the given region is in one of the supported regions.

This also makes https://github.com/agentsea/agentdesk/blob/main/agentdesk/vm/img.py#L20 this ami ID invalid. We will need to update https://github.com/agentsea/agentdesk/blob/main/agentdesk/vm/ec2.py#L69 to fetch the ami ID by image name, something like:

```py
import boto3

def find_ami_by_name(ami_name, region):
    ec2_client = boto3.client('ec2', region_name=region)
    
    response = ec2_client.describe_images(
        Filters=[
            {
                'Name': 'name',
                'Values': [ami_name]
            }
        ]
    )
    
    images = response.get('Images', [])
    
    if not images:
        print(f"No AMI found with the name: {ami_name}")
        return None
    
    # Assuming the first image in the response is the desired one
    ami = images[0]
    return ami
    ```

**Comment by nph4rd** [comment]
Ok - I can add those changes. Can we make it available on `ap-northeast-1` too? That's the region that's being used in the OP.

**Comment by nph4rd** [comment]
> We will need to update https://github.com/agentsea/agentdesk/blob/main/agentdesk/vm/ec2.py#L69

Interesting - it looks like the ID hasn't changed:

```
Found AMI with name: agentd-ubuntu-22.04-20240529073401 and ID: ami-06f3f5d8474ba4099
```

Anyway - opened https://github.com/agentsea/agentdesk/pull/26 for the validation.

**Comment by nph4rd** [comment]
Bumped `agentdesk` version in [6ad0397](https://github.com/agentsea/surfkit/pull/90/commits/6ad0397be531d8fcbeeb6ae873f9679f38314686)

**Comment by nph4rd** [comment]
Lint will fail until we can run the poetry lock with the new agentdesk version btw.

**Comment by pbarker** [comment]
`0.2.86` of agentdesk is now released, lets update the deps here and get this merged 🚢

**Comment by nph4rd** [comment]
done - @pbarker let's 🚢 :)

File: surfkit/cli/main.py [hunk 1]
```diff
@@ -141,6 +141,9 @@ def create_device(
     image: Optional[str] = typer.Option(
         None, help="The image to use for the desktop. Defaults to Ubuntu Jammy."
     ),
+    region: Optional[str] = typer.Option(
+        "us-east-1", help="AWS region. Defaults to 'us-east-1'."
```
Comment (pbarker) [inline surfkit/cli/main.py:145]: Sorry one last thing, can we add the options here?

Comment (nph4rd) [reply surfkit/cli/main.py:145]: ah - sure!

**Pull request merged**